### DISCUSSION TITLE
How is lidarr supposed to work correctly/ efficiently?

### DISCUSSION THREAD

**Discussion by basings** [RandomNinjaAtk/docker-lidarr-extended#198] [Q&A] [answered]
lidarr-extended runs since a few days and it mostly just keeps updating covers, sometimes searching indexers without getting results and rarely the log shows "downloading ... ". The quality of the final database looks very good but the whole process is very slow. Moreover, pressing on the search icon in lidarr does not appear to do anything. Could you please give me any hint on how it's supposed to work properly?

**Comment by RandomNinjaAtk** [comment]
Seems like your asking a general lidarr usage question, that would be more appropriate in lidarr support channels. The extended scripts are only able to monitored via log files...

Regular lidarr works just like the other Arr apps... (sonarr/radarr/readarr)

**Reply by basings** [reply]
I am familiar with lidarr, beets, deemix, deemon, lidarr-on-steroids, etc... I was just not familar with lidarr-extended (LE). I am still reading through the source code to understand it. I'm not yet sure how lidarr-extended is supposed to work properly.

I might have not expressed myself properly, sorry. I try again.

With default lidarr, you add an artist and look through an indexer for his releases. in the case of LE, the indexer is musicbrainz. Usually you set the indexer in the settings of lidarr, not in the case of LE where musicbrainz is fixed by the script. You may add an indexer but you can't remove musicbrainz.  After you added the artist and looked it up, you may send the request to a download client to get it. In the case of LE, the download client is deezer or tidal. Again, you can only specify it in the source code. Usually you can initiate the download process by pressing a  search button. Not in the case of lidarr-extended. It seems like you should not intervene with the script. Just set it up and let it do its magic. But it doesn't really do its magic. If I want to start the script I have to go into the container and start Audio.sh which runs everything. I can't tell it to search for artist xy in musicbrainz, then look it up in deezer and download it (and then postprocess it). According to the log, the script is always working, its always downloading a cover if you set it up to work automatically, unless it's sleeping for 15 minutes. But it doesn't always download something.

Sorry if it's too confusing and not on point, I just don't get the workflow of LE.

**Reply by RandomNinjaAtk** [reply]
If you set the parameters correctly, the script will autostart and run in background automatically...

Make sure you read through all the parameters carefully... see here: https://github.com/RandomNinjaAtk/docker-lidarr-extended#parameters

The idea is full automation, no user intervention needed.... other than the required authentication piece for tidal-dl.  Some people only use the scripts and don't bother with normal indexer/client because it works that well...

Edit: It does require some patience, but for the most part it just works... and does feel like magic...

**Reply by RandomNinjaAtk** [reply, marked as answer]
@basings My best advice for you would be to setup a test instance (clean library, clean root folder) of this using only Deemix client because the other client requires your input and is a bit more tricky.... Configure your folders naming and etc.

Add a single artist, and just wait/watch what happens. Logs are viewable from the Lidarr web gui, under "System" -> "Log Files". That way you can get a feel for it... The script can do quite a lot, such as:

Add artists related to your library artists
Add top artists from albums/tracks/artists lists
Tag files with beets, before and after import, when using the custom lidarr post processing scripts...
Download videos for your artists
Transcode the files to your preferred format
Add replaygain tags
Etc.... (see main page)
